src/connection/connection.js

```diff
@@ -33,17 +30,49 @@ class Connection {
    * @param {string} [properties.stat.encryption] connection encryption method identifier.
    */
   constructor ({ localAddr, remoteAddr, localPeer, remotePeer, newStream, close, getStreams, stat }) {
-    localAddr && assert(multiaddr.isMultiaddr(localAddr), 'localAddr must be an instance of multiaddr')
-    assert(PeerId.isPeerId(localPeer), 'localPeer must be an instance of peer-id')
-    assert(PeerId.isPeerId(remotePeer), 'remotePeer must be an instance of peer-id')
-    assert(typeof newStream === 'function', 'new stream must be a function')
-    assert(typeof close === 'function', 'close must be a function')
-    assert(typeof getStreams === 'function', 'getStreams must be a function')
-    assert(stat, 'connection metadata object must be provided')
-    assert(stat.direction === 'inbound' || stat.direction === 'outbound', 'direction must be "inbound" or "outbound"')
-    assert(stat.timeline, 'connection timeline object must be provided in the stat object')
-    assert(stat.timeline.open, 'connection open timestamp must be provided')
-    assert(stat.timeline.upgraded, 'connection upgraded timestamp must be provided')
+    if (localAddr && !multiaddr.isMultiaddr(localAddr)) {
```

Comment:
what do you think on getting a util function to validate all those parameters?

otherwise, this adds so much noise to the constructor

Reply:
I've pulled it out into a function.

Longer term using a validation library would be a much better option, or getting `libp2p` to validate things before passing them into constructors, or maybe not validating at all, given that we put little 'Transport compatible', 'Connection compatible', etc stickers all over our readmes (I'm kidding, we have tests for that).

If we go the validation library route, we should agree upfront which one to use so we don't have multiple modules that do the same job bloating bundle sizes.

Reply:
I've pulled it out into a function.

Longer term using a validation library would be a much better option, or getting libp2p to validate things before passing them into constructors, or maybe not validating at all, given that we put little 'Transport compatible', 'Connection compatible', etc stickers all over our readmes (I'm kidding, we have tests for that).

If we go the validation library route, we should agree upfront which one to use so we don't have multiple modules that do the same job bloating bundle sizes.